core/qthelper.cpp

```diff
@@ -927,17 +927,17 @@ int parseGasMixHE(const QString &text)
 QString get_dive_duration_string(timestamp_t when, QString hourText, QString minutesText, QString secondsText, bool isFreeDive)
 {
 	int hrs, mins, fullmins, secs;
-	mins = (when + 59) / 60;
+	mins = (when + 30) / 60;
 	fullmins = when / 60;
 	secs = when - 60 * fullmins;
 	hrs = mins / 60;
 
 	QString displayTime;
-	if (prefs.units.duration_units == units::ALWAYS_HOURS || (prefs.units.duration_units == units::MIXED && hrs)) {
+	if (isFreeDive) {
+		displayTime = QString("%1%2%3%4").arg(fullmins).arg(minutesText).arg(secs, 2, 10, QChar('0')).arg(secondsText);
+	} else if (prefs.units.duration_units == units::ALWAYS_HOURS || (prefs.units.duration_units == units::MIXED && hrs)) {
 		mins -= hrs * 60;
 		displayTime = QString("%1%2%3%4").arg(hrs).arg(hourText).arg(mins, 2, 10, QChar('0')).arg(hourText == ":" ? "" : minutesText);
-	} else if (isFreeDive) {
-		displayTime = QString("%1%2%3%4").arg(fullmins).arg(minutesText).arg(secs, 2, 10, QChar('0')).arg(secondsText);
```

Comment:
so the functional difference is...  in my code, people who selected ALWAYS_HOURS do indeed always get hours... and people who didn't pick hours but who had a freedive that lasted over an hour, also get that shown in hours. Both of those seemed /reasonable/ to me.
Your code always makes a freedive show minutes and seconds, even if the user explicitly asked to always be shown hours or does a dive that's longer than an hour. I'm open to arguments how that is better, but I'm doubtful.